### DISCUSSION TITLE
How to use consumer that has multiple subscriptions to pull from specific subjects

### DISCUSSION THREAD

**Discussion by rdkumavat1** [nats-io/nats.net#599] [General]
How to use consumer to pull from specific subject, consumer has subscription on more than one subject.
example with code will be more helpful..

Also please suggest how to use - NotificationHandler to call a method to process message, when new message will be generated.

Thanks

**Comment by rickdotnet** [comment]
I think you're looking for [FilterSubjects](https://docs.nats.io/nats-concepts/jetstream/consumers#filtersubjects)

**Reply by rdkumavat1** [reply]
No, I think, Let me explain with an example - 

I have a stream say -  factory-events 
with subjects factory-events.A and factory-events.B
having a consumer C1 that has subscription on both the subjects - factory-events.A and factory-events.B

But using the same consumer I want to pull or consume messages from the subject that will be decided on the method input parameter

For ex - ConsumeMsg(string subjectname)

Then, this ConsumeMsg method should use consumer C1 to pull only from the subject name provided in the subjectName parameter of this method.

Thanks

**Reply by rickdotnet** [reply]
Please share what you have and I can attempt to help with your code. But, this seems more like an application design concern. The client is configured to consume both messages. If you're wanting receive both, but consume one, you could simply inspect the subject and decide what you want to do from there. In the [example here](https://github.com/ConnectEverything/nats-by-example/blob/983737bf545a38626954cf1d2ba51904929f040e/examples/jetstream/pull-consumer/dotnet2/Main.cs#L53), you'd have the `msg.Subject` to make decisions on.
